src/components/Widgets/EventsWidget.tsx

```diff
@@ -49,63 +60,140 @@ const EmptyStateBellIcon: React.FunctionComponent = () => (
-        <EmptyState
-          titleText={
-            <Title headingLevel="h4" size="lg">
-              {intl.formatMessage(messages.noFiredEventsTitle)}
-            </Title>
-          }
-          icon={EmptyStateBellIcon}
-          variant={EmptyStateVariant.full}
+  const handlePageChange = useCallback(
+    (
+      event: React.MouseEvent | React.KeyboardEvent | MouseEvent,
+      newPage: number
+    ) => {
+      onSetPage(event, newPage);
+      fetchNotifications(newPage, perPage);
+    },
+    [onSetPage, fetchNotifications, perPage]
+  );
+
+  const handlePerPageChange = useCallback(
+    (
+      event: React.MouseEvent | React.KeyboardEvent | MouseEvent,
+      newPerPage: number
+    ) => {
+      onPerPageSelect(event, newPerPage);
+      onSetPage(event, 1);
+      fetchNotifications(1, newPerPage);
+    },
+    [onPerPageSelect, onSetPage, fetchNotifications]
+  );
+
+  const emptyState = useMemo(
+    () => (
+      <EmptyState
+        titleText={
+          <Title headingLevel="h4" size="lg">
+            {intl.formatMessage(messages.noFiredEventsTitle)}
+          </Title>
+        }
+        icon={EmptyStateBellIcon}
+        variant={EmptyStateVariant.full}
+      >
+        <EmptyStateBody>
+          <Stack>
+            <StackItem>
+              {intl.formatMessage(messages.noFiredEventsDescription)}
+            </StackItem>
+          </Stack>
+        </EmptyStateBody>
+        <Button
+          component="a"
+          variant={ButtonVariant.secondary}
+          className="pf-v5-u-mt-lg"
+          href="settings/notifications"
         >
-          <EmptyStateBody>
-            <Stack>
-              <StackItem>
-                {intl.formatMessage(messages.noFiredEventsDescription)}
-              </StackItem>
-            </Stack>
-          </EmptyStateBody>
-          <Button
-            component="a"
-            variant={ButtonVariant.secondary}
-            className="pf-v5-u-mt-lg"
-            href="settings/notifications"
-          >
-            {intl.formatMessage(messages.manageEvents)}
-          </Button>
-        </EmptyState>
-      ) : (
-        <Table aria-label="Events widget table" variant={TableVariant.compact}>
+          {intl.formatMessage(messages.manageEvents)}
+        </Button>
+      </EmptyState>
+    ),
+    [intl]
+  );
+
+  const loadingStateHeader = useMemo(
+    () => (
+      <SkeletonTableHead
+        columns={[
+          intl.formatMessage(messages.event),
+          intl.formatMessage(messages.service),
+          intl.formatMessage(messages.date),
+        ]}
+      />
+    ),
+    [intl]
+  );
+
+  const loadingStateBody = useMemo(
+    () => <SkeletonTableBody rowsCount={perPage} columnsCount={3} />,
+    [perPage]
+  );
+
+  return (
+    <DataView
+      activeState={
+        loading ? 'loading' : notifications.length === 0 ? 'empty' : undefined
+      }
+    >
+      <Table aria-label="Events widget table" variant={TableVariant.compact}>
```

Comment:
I think we should be using DataViewTable for this and then we can pass loading and empty states as props for it.

Reply:
Ah good catch!